vitrina/datasets/views.py

```diff
@@ -14,3 +20,168 @@ class DatasetListView(ListView):
+        query = self.request.GET.urlencode()
+
+        selected_status = self.get_selected_value('status', is_int=False)
+        selected_organization = self.get_selected_value('organization')
+        selected_categories = self.get_selected_value('category', multiple=True)
+        selected_frequency = self.get_selected_value('frequency')
+        selected_tags = self.get_selected_value('tags', multiple=True, is_int=False)
+        selected_date_from = self.get_selected_value('date_from', is_int=False)
+        selected_date_to = self.get_selected_value('date_to', is_int=False)
+
+        related_categories = get_related_categories(selected_categories)
+
+        # after task #118 tags filter should be redone
+        tags = Dataset.public.exclude(tags="").exclude(tags__isnull=True).values_list('tags', flat=True)
+        tag_list = []
+        for dataset_tags in tags:
+            dataset_tags = dataset_tags.replace(" ", "").split(",")
+            tag_list.extend(dataset_tags)
+        tag_list = list(set(tag_list))
+
+        related_tag_list = []
+        if selected_tags:
+            related_tags = filtered_queryset.exclude(tags="").exclude(tags__isnull=True).values_list('tags', flat=True)
+            related_tag_list = []
+            for dataset_tags in related_tags:
+                dataset_tags = dataset_tags.replace(" ", "").split(",")
+                related_tag_list.extend(dataset_tags)
+            related_tag_list = list(set(related_tag_list))
+            related_tag_list.sort()
+
+        status_counts = {}
+        for status in Dataset.FILTER_STATUSES.keys():
+            status_counts[status] = filter_by_status(filtered_queryset, status).count()
+
+        category_counts = {}
+        if selected_categories:
+            for category in related_categories:
+                try:
+                    category = Category.objects.get(pk=category)
+                except ObjectDoesNotExist:
+                    category_counts[category] = 0
+                    continue
+                count = filtered_queryset.filter(category=category).count()
+                children = category.get_descendants()
+                for ch in children:
+                    count += filtered_queryset.filter(category=ch).count()
+                category_counts[category.pk] = count
+        else:
+            for category in Category.objects.all():
+                count = filtered_queryset.filter(category=category).count()
+                children = category.get_descendants()
+                for ch in children:
+                    count += filtered_queryset.filter(category=ch).count()
+                category_counts[category.pk] = count
+
+        extra_context = {
+            'num_found': filtered_queryset.count(),
+            'status_filters': [{
+                'key': key,
+                'title': value,
+                'query': "?%s%sstatus=%s" % (query, "&" if query else "", key),
```

Comment:
It looks, that new filters are always appended on top of existing ones. I think, better way to handle this, is to have `query` as a dict and update it, replacing or appending previously set values. Then you can simply [urlencode](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlencode) dict Also, please move this url building functionality into a separate function in `helpers.py` and add tests for it. The end result could look like this:

```suggestion
                'query': get_filter_url(query, 'status', key, append=False),
```